tests/test_minimal/test_tools/aws_tools.py

```diff
@@ -156,3 +180,120 @@ def test_submit_aws_batch_job_with_dependencies():
+
+    job_name = "test_submit_aws_batch_job_with_efs"
+    docker_image = "ubuntu:latest"
+    date = datetime.datetime.now().date().strftime("%y%m%d")
+    commands = ["touch", f"/mnt/efs/test_{date}.txt"]
+
+    # TODO: to reduce costs even more, find a good combinations of memory/CPU to minimize size of instance
+    efs_volume_name = f"test_neuroconv_batch_with_efs_{date}"
+    info = submit_aws_batch_job(
+        job_name=job_name,
+        docker_image=docker_image,
+        commands=commands,
+        efs_volume_name=efs_volume_name,
+    )
+
+    # Wait for AWS to process the job
+    time.sleep(60)
+
+    job_id = info["job_submission_info"]["jobId"]
+    job = None
+    max_retries = 10
+    retry = 0
+    while retry < max_retries:
+        job_description_response = batch_client.describe_jobs(jobs=[job_id])
+        assert job_description_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+        jobs = job_description_response["jobs"]
+        assert len(jobs) == 1
+
+        job = jobs[0]
+
+        if job["status"] in _RETRY_STATES:
+            retry += 1
+            time.sleep(60)
+        else:
+            break
+
+    # Check EFS specific details
+    efs_volumes = efs_client.describe_file_systems()
+    matching_efs_volumes = [
+        file_system
+        for file_system in efs_volumes["FileSystems"]
+        for tag in file_system["Tags"]
+        if tag["Key"] == "Name" and tag["Value"] == efs_volume_name
+    ]
+    assert len(matching_efs_volumes) == 1
+    efs_volume = matching_efs_volumes[0]
+    efs_id = efs_volume["FileSystemId"]
+
+    # Check normal job completion
+    assert job["jobName"] == job_name
+    assert "neuroconv_batch_queue" in job["jobQueue"]
+    assert "fs-" in job["jobDefinition"]
+    assert job["status"] == "SUCCEEDED"
+
+    status_tracker_table_name = "neuroconv_batch_status_tracker"
+    table = dynamodb_resource.Table(name=status_tracker_table_name)
+    table_submission_id = info["table_submission_info"]["id"]
+
+    table_item_response = table.get_item(Key={"id": table_submission_id})
+    assert table_item_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+    table_item = table_item_response["Item"]
+    assert table_item["job_name"] == job_name
+    assert table_item["job_id"] == job_id
+    assert table_item["status"] == "Job submitted..."
+
+    table.update_item(
+        Key={"id": table_submission_id},
+        AttributeUpdates={"status": {"Action": "PUT", "Value": "Test passed - cleaning up..."}},
+    )
+
+    # Cleanup EFS after testing is complete - must clear mount targets first, then wait before deleting the volume
+    # TODO: cleanup job definitions? (since built daily)
+    mount_targets = efs_client.describe_mount_targets(FileSystemId=efs_id)
+    for mount_target in mount_targets["MountTargets"]:
+        efs_client.delete_mount_target(MountTargetId=mount_target["MountTargetId"])
+
+    time.sleep(60)
+    efs_client.delete_file_system(FileSystemId=efs_id)
```

Comment:
The the future I plan to be more elegant with pytest cleanup, possibly via `pytest_sessionfinish` in a `conftest.py`

It might also cleanup any job definitions that start with 'test_neuroconv'